extensions/wikia/AdEngine/js/recovery/helper.js

```diff
@@ -0,0 +1,75 @@
+define('ext.wikia.adEngine.recovery.helper', [
+	'ext.wikia.adEngine.adContext',
+	'wikia.document',
+	'wikia.lazyqueue',
+	'wikia.log',
+	require.optional('ext.wikia.adEngine.provider.gpt.sourcePointTag')
+], function (
+	adContext,
+	doc,
+	lazyQueue,
+	log,
+	SourcePointTag
+) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.recovery.helper',
+		context = adContext.getContext(),
+		slotsToRecover = [],
+		onBlockingEventsQueue = [];
+
+	function initEventQueue() {
+		lazyQueue.makeQueue(onBlockingEventsQueue, function (eventHandler) {
+			doc.addEventListener('sp.blocking', function () {
+				eventHandler();
+			});
+		});
+		onBlockingEventsQueue.start();
```

Comment:
What do you think about making lazy queue of callbacks instead of new EventListeners? We could add only callbacks to queue and start it once `sp.blocking` event is dispatched. In this case later added callbacks would be executed immediately after pushing to queue. Other benefit is that we would have only one listener for `sp.blocking` event.

**Other note if you agree:**
~~After creating this kind of queue we should check whether `sp.blocking` is already dispatched - if it is we should start() queue immediately, otherwise we can set up start() as a callback for `sp.blocking` listener.~~
I see that queue is initialised before detection, so we can just start queue on `sp.blocking` event.